### PR TITLE
[WIP] Fixed reorder to 0

### PR DESCRIPTION
**Pull request by dantleech**
This PR fixes the scenario where a node needs to be repositoned to 0. Need to add tests.

### CODE REVIEW

File: Repository/PhpcrRepository.php [hunk 1]
```diff
@@ -140,7 +140,9 @@ public function reorderNode($sourcePath, $position)
         $parent = $node->getParent();
         $nodeNames = $parent->getNodeNames();
 
-        if (isset($nodeNames[$position + 1])) {
+        if ($position == 0) {
+            $parent->orderBefore($node->getName(), $nodeNames[$position]);
```
Comment (dantleech) [inline Repository/PhpcrRepository.php:144]: Other edge cases here? $nodeNames[0] should always be set.

**Comment by dantleech** [comment]
This functionality is covered by the functional tests in the bundle, merging.
